files/image_config/monit/monitoring_containers

```diff
@@ -0,0 +1,116 @@
+              if the device has Multi-ASIC, this function will generate list by determining the
+              value of field 'has_global_scope', the number of ASICs and the value of field of
+              'has_per_asic_scope'. If the device has single ASIC, just put the container name into
+              the list.
+    @return:  A set which contains the expected running containers.
+    """
+    config_db = swsssdk.ConfigDBConnector()
+    config_db.connect()
+    feature_table = config_db.get_table("FEATURE")
+
+    expected_running_containers = set()
+
+    for container_name in feature_table.keys():
+        if feature_table[container_name]["state"] != "disabled":
+            if multi_asic.is_multi_asic():
+                if feature_table[container_name]["has_global_scope"] == "True":
+                    expected_running_containers.add(container_name)
+                if feature_table[container_name]["has_per_asic_scope"] == "True":
+                    num_asics = multi_asic.get_num_asics()
+                    for asic_id in range(num_asics):
+                        expected_running_containers.add(container_name + str(asic_id))
+            else:
+                expected_running_containers.add(container_name)
+
+    return expected_running_containers
+
+
+def get_current_running_containers():
+    """
+    @summary: This function will get the current running container list by analyzing the
+              output of command `docker ps`.
+    @return:  A set which contains the current running contianers.
+    """
+    running_containers = set()
+
+    command = "docker ps"
+    command_stdout = get_command_result(command)
+    for line in command_stdout:
+        if line and "CONTAINER ID" not in line:
+            running_containers.add(line.split()[-1].strip())
+
+    return running_containers
+
+
+def monitoring_containers_and_alerting():
+    """
+    @summary: This function will compare the difference between the current running containers
+              and the containers which were expected to run. If containers which were exepcted
+              to run but not running, then an alerting message will be written into syslog.
+    """
+    expected_running_containers = get_expected_running_containers()
+    current_running_containers = get_current_running_containers()
+
+    not_running_containers = expected_running_containers.difference(current_running_containers)
+
+    if not_running_containers:
+        log_message = ""
+        for container_name in not_running_containers:
+            if not log_message:
+                log_message = container_name
+            else:
+                log_message += ", " + container_name
+
+        if len(not_running_containers) > 1:
+            print("Containers '{}' were not running.".format(log_message))
+        else:
+            print("Container '{}' was not running.".format(log_message))
+
+        sys.exit(4)
+
+
+def main():
+    monitoring_containers_and_alerting()
```

Comment:
This is pointless. Suggest removing this and simply renaming `monitoring_containers_and_alerting()` to `main()`

Reply:
Great suggestions! Updated.